daml-lf/validation/src/main/scala/com/digitalasset/daml/lf/validation/Serializability.scala

```diff
@@ -37,7 +37,8 @@ private[validation] object Serializability {
 
     def checkType(): Unit = checkType(typeToSerialize)
 
-    def checkType(typ0: Type): Unit = typ0 match {
+    def checkType(typ0: Type): Unit = typ0 match { //TODO: fix the exhaustiveness checking
```

Comment:
I don’t understand this comment, aren’t you already handling the case that you added?

Reply:
This comments is noting that the scala exhaustivity checking is not being done on this match expression. Which is sad. And it would be nice to get it fixed.

Reply:
Some more details: The match expression makes use of `unapply` methods defined for `ParametricType[12]` in 
daml-lf/language/src/main/scala/com/digitalasset/daml/lf/language/Util.scala

Sadly, unapply methods cause the scala match-exhaustivity checker to get disabled (without any warning).

Reply:
scala is great …

Reply:
Maybe we should remove the TODO, but I think it worth keeping the comment that there is no exhaustivity check for these two match expressions.

Reply:
Happy to keep the comment. I would suggest to turn it into a non-TODO comment and make it a bit more explicit that this is about the scala exhaustiveness checker not checking this match rather and not this match not being exhaustive (at least at this point :slightly_smiling_face:)

Reply:
I made a separate PR to fix this case of lost exhaustivity checking. #4139